gnomad_qc/v3/annotations/generate_qc_annotations.py

```diff
@@ -55,24 +82,35 @@ def compute_info() -> hl.Table:
             hl.agg.group_by(
                 get_adj_expr(mt.LGT, mt.GQ, mt.DP, mt.LAD),
                 hl.agg.sum(
-                    mt.LGT.one_hot_alleles(mt.LA.map(lambda x: hl.str(x)))[mt.LA.index(ai)]
-                )
-            )
+                    mt.LGT.one_hot_alleles(mt.LA.map(lambda x: hl.str(x)))[
+                        mt.LA.index(ai)
+                    ]
+                ),
+            ),
         ),
-        mt['alt_alleles_range_array']
+        mt.alt_alleles_range_array,
     )
 
     # Then, for each non-ref allele, compute
     # AC as the adj group
     # AC_raw as the sum of adj and non-adj groups
     info_expr = info_expr.annotate(
         AC_raw=grp_ac_expr.map(lambda i: hl.int32(i.get(True, 0) + i.get(False, 0))),
-        AC=grp_ac_expr.map(lambda i: hl.int32(i.get(True, 0)))
+        AC=grp_ac_expr.map(lambda i: hl.int32(i.get(True, 0))),
     )
 
-    info_ht = mt.select_rows(
-        info=info_expr
-    ).rows()
+    # Annotating raw MT with pab max
+    info_expr = info_expr.annotate(
+        AS_pab_max=hl.agg.array_agg(
+            lambda ai: hl.agg.filter(
+                mt.LA.contains(ai) & mt.LGT.is_het(),
+                hl.agg.max(hl.binom_test(mt.LAD[1], hl.sum(mt.LAD), 0.5, "two-sided")),
```

Comment:
Yup, I think you are right about both of those points. I will add in the fix. Thank you for catching, I think it has been through like 4 reviews and wasn't caught before. I will have to change the UKBB code too, hopefully this improves the RFs

Reply:
OK, so there does need to be a change to the code, but it isn’t the exact same problem you flagged. I thought that the code as its was actually fine after looking back over the documentation for the sparse MT (and because I filter to only genotypes that include `ai`), but I made the incorrect assumption that in a single individual there would only be reads contributing to two alleles. In that case LAD would only be of length 2 (since it only includes alleles with reads) and the current code would work, but this is not always the case. The denominator is actually fine as is, but the numerator needs to be adjusted to account for possible reads that contain another allele other than ref or the majority alt. From my test on all of chr22, this error impacts ~0.1% of variants (prior to any filtering), so likely won’t have a huge impact on the RF, but worth checking. I have put in the fix and compared to the pab_max calculated on the split table using AD, there are now 0 differences.

Reply:
Great! I'm still a little fuzzy as to why this code works -- can you explain what `LAD` includes and why it should be fine on multi-allelics?

Reply:
LAD only includes info for alleles that are present in that sample, so there are no 0's. It is the same length as LA. Here is the example on Hail for a multi-allelic variant:
<img width="371" alt="Screen Shot 2020-12-16 at 9 41 51 AM" src="https://user-images.githubusercontent.com/33063077/102378366-f560eb00-3f82-11eb-8fb1-14fed2b8189a.png">